### ISSUE TITLE
L.Browser.pointer value is wrong

### ISSUE THREAD

**Issue by johnd0e** [Leaflet/Leaflet#6977]
Here are both docs and code:
https://github.com/Leaflet/Leaflet/blob/d1a1e97b8290f642eb677284af53c2db64199a76/src/core/Browser.js#L94-L96

It is clearly seen that `pointer` value will be `false` in webkit despite docs statement.

The issue appeared after #6855.

I understand that PR purpose, but still insist that proposed fix was not proper.
1. `webkit` is not iOS-specific property.
   `safari` is more close, but still, that was not proper place to make that fix.
2. Perhaps this place is better to check for `safari`:
   https://github.com/Leaflet/Leaflet/blob/d1a1e97b8290f642eb677284af53c2db64199a76/src/map/handler/Map.Tap.js#L131-L136
   At least change here would be more local.
3. Enabling `tap` handler indeed fixes issue with `contextmenu` event (though not completely: #6865), but it also has unwanted sideeffects.
   Sample to reproduce issues involves `Leaflet.draw`, which is old, unsupported,and need to be fixed itself.
But if it is relevant - I can setup a fiddle.

**Comment by IvanSanchez** [comment]
This sounds sensible (reverting #6855 and adding an extra check at `Map.Tap.js`), but I don't have any iOS devices to test this and #6865 and related issues. I'll have to trust whoever sends a PR having tested out in different platforms.

**Comment by alcalin** [comment]
This is also making **touch**  false, and is bringing along visual changes because of the  'leaflet-touch' class missing from the map container ex.
```
.leaflet-touch .leaflet-bar a {
    width: 30px;
    height: 30px;
    line-height: 30px;
}
```

**Comment by johnd0e** [comment]
> IvanSanchez changed the title L.Browser.pointer value is wrong for iOS safari

Title of this issue change is wrong, as it is related to any webkit-based browser, including desktop Chrome and android webview.

**Comment by johnd0e** [comment]
BTW, could someone tell me if that `contextmenu` issue is some well known iOS limitation, or leaflet own bug?

I'm afraid that fixing it via Map.Tap handler we are only masking some more deep issue.

**Comment by johnd0e** [comment]
Ok, I've investigated iOS issue a bit, and here is the summary:
- `L.Browser.pointer` has clearly documented meaning, and other properties depend on it as seen https://github.com/Leaflet/Leaflet/blob/master/src/core/Browser.js
   So it's terribly wrong to change it's value for unrelated reason, that's why #6855 must be reverted in any case.

I agree that we still need to fix iOS issues, but in right way:
1. Clearly define the origin of iOS issue.
   As I've found, it is rather simple: `contextmenu` is not fired.
   It is currently solved by `L.Map.Tap`, but that handler also has several unrelated sideeffects (#6980), so ideally it'd be to implement separate handler with single feature - firing `contextmenu`.
2. Implement proper detection of conditions when we need to apply the fix.
   It's obvious that relying on `L.Browser.pointer` and `L.Browser.webkit` is terribly wrong.
   Currently best I can to propose is either check `L.Browser.safari` or search for `'iOS'` substring in `navigator.userAgent` (need to check which suits better).
   Maybe someone can point to better way later.
